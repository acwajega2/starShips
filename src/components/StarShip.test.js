import React from 'react';
import { shallow, mount } from '../Enzyme';
import StarShip from './StarShip';
import renderer from 'react-test-renderer';



describe("Testing the StarShip Component", () =>{


  //------Doing SnapShot Testing---------------------------

    it("Matches the previous snap shot", () => {
        const sampleStarShip = {
            name: "shipA",
            crew: 23,
            passengers: 10,
            hyperdrive_rating : 2.4
            
      }
        const component = renderer.create(<StarShip key={2} starShip={sampleStarShip} />).toJSON();
        expect(component).toMatchSnapshot()
        
  });
  //------- End of Snapshot Testing--------------------
    
//----- Testing if the StarShip Component Initializes Properly
    it("Should Initialize properly", () => {
        const sampleStarShip = {
            name: "shipA",
            crew: 23,
            passengers: 10,
            hyperdrive_rating : 2.4
            
      }
        shallow(<StarShip key={2} starShip={sampleStarShip} />)
    })

    
//---------End of Initialization Testing
    
    
    
    
    

    
  
 

})