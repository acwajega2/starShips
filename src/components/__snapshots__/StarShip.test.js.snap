exports[`Testing the StarShip Component Matches the previous snap shot 1`] = `
<div
  className="mx-auto ">
  <div
    className="ant-card text-left ant-card-bordered ant-card-hoverable">
    <div
      className="ant-card-head"
      style={Object {}}>
      <div
        className="ant-card-head-wrapper">
        <div
          className="ant-card-head-title">
          shipA
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}>
      <p>
        Crew : 
        23
      </p>
      <p>
        Passengers : 
        10
      </p>
      <p>
        Hyperdrive Class :
        <div
          className="ant-progress ant-progress-line ant-progress-status-active ant-progress-default">
          <div>
            <div
              className="ant-progress-outer">
              <div
                className="ant-progress-inner">
                <div
                  className="ant-progress-bg"
                  style={
                    Object {
                      "background": undefined,
                      "borderRadius": "",
                      "height": 8,
                      "width": "40%",
                    }
                  } />
              </div>
            </div>
          </div>
        </div>
      </p>
    </div>
  </div>
</div>
`;

exports[`Testing the StarShip Component Should Render Properly 1`] = `
<div
  className="mx-auto ">
  <div
    className="ant-card text-left ant-card-bordered ant-card-hoverable">
    <div
      className="ant-card-head"
      style={Object {}}>
      <div
        className="ant-card-head-wrapper">
        <div
          className="ant-card-head-title">
          shipA
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}>
      <p>
        Crew : 
        23
      </p>
      <p>
        Passengers : 
        10
      </p>
      <p>
        Hyperdrive Class :
        <div
          className="ant-progress ant-progress-line ant-progress-status-active ant-progress-default">
          <div>
            <div
              className="ant-progress-outer">
              <div
                className="ant-progress-inner">
                <div
                  className="ant-progress-bg"
                  style={
                    Object {
                      "background": undefined,
                      "borderRadius": "",
                      "height": 8,
                      "width": "40%",
                    }
                  } />
              </div>
            </div>
          </div>
        </div>
      </p>
    </div>
  </div>
</div>
`;

exports[`Testing the StarShip Component Should Run Properly 1`] = `
<div
  className="mx-auto ">
  <div
    className="ant-card text-left ant-card-bordered ant-card-hoverable">
    <div
      className="ant-card-head"
      style={Object {}}>
      <div
        className="ant-card-head-wrapper">
        <div
          className="ant-card-head-title">
          shipA
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}>
      <p>
        Crew : 
        23
      </p>
      <p>
        Passengers : 
        10
      </p>
      <p>
        Hyperdrive Class :
        <div
          className="ant-progress ant-progress-line ant-progress-status-active ant-progress-default">
          <div>
            <div
              className="ant-progress-outer">
              <div
                className="ant-progress-inner">
                <div
                  className="ant-progress-bg"
                  style={
                    Object {
                      "background": undefined,
                      "borderRadius": "",
                      "height": 8,
                      "width": "40%",
                    }
                  } />
              </div>
            </div>
          </div>
        </div>
      </p>
    </div>
  </div>
</div>
`;
